### PR TITLE
Plant infrastructure for LIMA_PROTOCOL_VERSION

### PR DESCRIPTION
**Pull request by gorillainduction**
https://www.pivotaltracker.com/story/show/166205915

### CODE REVIEW

File: rebar.config [hunk 1]
```diff
@@ -137,6 +137,10 @@
                    {copy, "data/aecore/.fortuna/accounts.json", "data/aecore/.fortuna/accounts.json"},
                    {copy, "data/aecore/.fortuna/accounts_uat.json", "data/aecore/.fortuna/accounts_uat.json"},
                    {copy, "data/aecore/.fortuna/accounts_test.json", "data/aecore/.fortuna/accounts_test.json"},
+                   {mkdir, "data/aecore/.lima"},
```
Comment (lucafavatella) [inline rebar.config]: I wonder if these packaging lines can be split in separate line.

File: ci/appveyor/package.bat [hunk 2]
```diff
@@ -54,6 +54,7 @@ rem Copy genisis and hard-fork account migrations into top-level data folder
 %BASH_BIN% -lc "cp -R \"%RELEASE_PATH%/data/aecore/.genesis\" \"%PACKAGE_PATH%/aeternity-windows-w64/data/aecore\""
 %BASH_BIN% -lc "cp -R \"%RELEASE_PATH%/data/aecore/.minerva\" \"%PACKAGE_PATH%/aeternity-windows-w64/data/aecore\""
 %BASH_BIN% -lc "cp -R \"%RELEASE_PATH%/data/aecore/.fortuna\" \"%PACKAGE_PATH%/aeternity-windows-w64/data/aecore\""
+%BASH_BIN% -lc "cp -R \"%RELEASE_PATH%/data/aecore/.lima\" \"%PACKAGE_PATH%/aeternity-windows-w64/data/aecore\""
```
Comment (lucafavatella) [inline ci/appveyor/package.bat]: I see this is re packaging too.

File: rebar.config (hunk 1 above)
Comment (gorillainduction) [reply rebar.config]: Actually, they are needed for the tests to pass. So, no.

File: apps/aecore/test/aec_test_utils.erl [hunk 3]
```diff
@@ -509,7 +509,8 @@ signed_spend_tx(ArgsMap) ->
 copy_forks_dir(SourceRelDir, DestRelDir) ->
     copy_fork_dir(SourceRelDir, DestRelDir, ?ROMA_PROTOCOL_VSN),
     copy_fork_dir(SourceRelDir, DestRelDir, ?MINERVA_PROTOCOL_VSN),
-    copy_fork_dir(SourceRelDir, DestRelDir, ?FORTUNA_PROTOCOL_VSN).
+    copy_fork_dir(SourceRelDir, DestRelDir, ?FORTUNA_PROTOCOL_VSN),
+    copy_fork_dir(SourceRelDir, DestRelDir, ?LIMA_PROTOCOL_VSN).
```
Comment (lucafavatella) [inline apps/aecore/test/aec_test_utils.erl]: I see this is for integration tests. Can stay.

File: apps/aecore/test/aec_block_fork_tests.erl [hunk 4]
```diff
@@ -65,7 +65,9 @@ load_files_smoke_test_() ->
                 (?MINERVA_PROTOCOL_VSN) ->
                     DataAecoreDir ++ ".minerva/" ++ File;
                 (?FORTUNA_PROTOCOL_VSN) ->
-                    DataAecoreDir ++ ".fortuna/" ++ File
+                    DataAecoreDir ++ ".fortuna/" ++ File;
+                (?LIMA_PROTOCOL_VSN) ->
+                    DataAecoreDir ++ ".lima/" ++ File
```
Comment (lucafavatella) [inline apps/aecore/test/aec_block_fork_tests.erl]: This can stay - as lines 82-83 do not trigger this ;)

**Comment by lucafavatella** [comment]
@gorillainduction You may want to cherry-pick some CI fixes from https://github.com/aeternity/aeternity/pull/2467

**Comment by lucafavatella** [comment]
From https://circleci.com/gh/aeternity/aeternity/36699#tests/containers/0
```
===> Command 'ct-latest' not found
make: *** [smoke-test-run] Error 1
```

**Comment by lucafavatella** [comment]
From docker_system_smoke_tests job https://circleci.com/gh/aeternity/aeternity/36815 after having added `ct-latest` target:
```
===> Command 'ct-latest' not found
```

So it seems that `machine_build_cache_key` needs bump (v6 -> v7).

**Comment by lucafavatella** [comment]
From test_fortuna job https://circleci.com/gh/aeternity/aeternity/36805
```
%%% aehttp_contracts_SUITE ==> fate.environment_contract_fate: FAILED
%%% aehttp_contracts_SUITE ==> {max_reached,
    [{aecore_suite_utils,mine_blocks_until_txs_on_chain,4,
```

So it seems that that test shall not run for Fortuna.

**Comment by lucafavatella** [comment]
CI failure seems [unrelated to PR](https://www.pivotaltracker.com/story/show/163696685/comments/202976669).

**Comment by lucafavatella** [comment]
Updated wiki:
[ct -> ct-latest](https://github.com/aeternity/aeternity/wiki/Integration-Testing/_compare/416209d0e235e632ccf18629b508ddeba0c007fc...be19dd97f737d88e61e5545be26cef2056aa9bbf)
[eunit -> eunit-latest](https://github.com/aeternity/aeternity/wiki/Unit-Testing/_compare/cc0aa15ca407cf2ca1f5d6e4bd8552e155640ff9...74fd030edbdbbcecbf402e5671128876787c54af)

**Pull request merged**